cmd/podman/oci-trace-hook/trace.go

```
@@ -0,0 +1,238 @@
+	log.Println(os.Args)
+	terminate := flag.Bool("e", false, "send SIGINT to floating process")
+	runBPF := flag.Int("r", 0, "-r <pid of the container>")
+	flag.Parse()
+
+	if *runBPF > 0 {
+		syscalls := make(calls, 303)
+		pid := *runBPF
+		src := strings.Replace(source, "PARENT_PID", strconv.Itoa(pid), -1)
+		m := bpf.NewModule(src, []string{})
+		defer m.Close()
+
+		tracepoint, err := m.LoadTracepoint("enter_trace")
+		if err != nil {
+			log.Println(err)
+		}
+
+		if err := m.AttachTracepoint("raw_syscalls:sys_enter", tracepoint); err != nil {
+			log.Println("unable to load tracepoint")
+		}
+
+		log.Println("tracepoint attached")
+
+		table := bpf.NewTable(m.TableId("events"), m)
+		channel := make(chan []byte)
+		perfMap, err := bpf.InitPerfMap(table, channel)
+		if err != nil {
+			log.Println("unable to init perf map")
+		}
+
+		sig := make(chan os.Signal, 1)
+		signal.Notify(sig, os.Interrupt, os.Kill)
+		rsc := false
+		go func() {
+			var e event
+			for {
+				data := <-channel
+				err := binary.Read(bytes.NewBuffer(data), binary.LittleEndian, &e)
+				if err != nil {
+					log.Printf("failed to decode received data '%s': %s\n", data, err)
+					continue
+				}
+				// comm := (*C.char)(unsafe.Pointer(&e.Command))
+				// log.Printf("Pid : %d, Syscall_ID : %d, , Command : %q", e.Pid, e.Id, C.GoString(comm))
+				name := getName(e.ID)
+				if name == "seccomp" {
+					rsc = true
+					continue
+				}
+				if rsc {
+					syscalls[name]++
+				}
+			}
+		}()
+		log.Println("PerfMap Start")
+		perfMap.Start()
+		<-sig
+		log.Println("PerfMap Stop")
+		perfMap.Stop()
+		generateProfile(syscalls, os.TempDir()+"/profile.json")
+	} else if *terminate {
+		log.Println("terminator")
+		f, err := ioutil.ReadFile("pid")
+
+		if err != nil {
+			log.Println(err)
+		}
+
+		log.Println("file opened")
+		Spid := string(f)
+
+		log.Println(Spid)
+		pid, _ := strconv.Atoi(Spid)
+		p, _ := os.FindProcess(pid)
+		p.Signal(os.Interrupt)
+
+	} else {
```

What is this third option? Generating BPF and Terminating make sense, they have CLI arguments, but this one isn't documented at all

---

This is still valid. Please add a new flag for that. It is confusing that there is silent default. If non of the flags is specified, we can error out.